app/components/upload-csv-popup.js

```diff
@@ -0,0 +1,431 @@
+                        <span className={classes.orText}>or</span> <span className={classes.browseText}>BROWSE</span>
+                    </div>
+                </FileDrop>
+            </label>
+        )
+    }
+
+    const renderSelectedFileBox = () => {
+        return (
+            <div className={classes.fileBox}>
+                <div className={classes.selectedFileText}>{selectedFile.name}</div>
+                <button type='button' className={classes.removeFileButton} onClick={() => handleRemoveFile()}>
+                    Remove File
+                </button>
+            </div>
+        )
+    }
+
+    return (
+        <div
+            id='upload-csv-popup'
+            className={cx(className, classes.popup)}
+            style={{ visibility: visible ? 'visible' : 'hidden', ...style }}
+        >
+            <div className={classes.innerPopup}>
+                <div className={classes.popupTop}>
+                    <span className={classes.provideText}>Provide Election Table</span>
+                    <div className={classes.checkSampleRow}>
+                        <div>Upload CSV File</div>
+                        <a href='/assets/csv/sample-candidate-table.csv' className={classes.checkSampleText}>
+                            Check Sample &nbsp;
+                            <ExternalLinkSvg className={classes.externalLinkIcon} />
+                        </a>
+                    </div>
+                    {renderErrors()}
+                    {selectedFile ? renderSelectedFileBox() : renderDropFile()}
+                    <input
+                        type='file'
+                        id='file-select'
+                        key='1'
+                        data-testid='file-select-input'
+                        style={{ display: 'none' }}
+                        ref={fileInputEl}
+                        onChange={onFileChange}
+                    />
+                </div>
+                <div className={classes.popupButtons}>
+                    <button type='button' className={cx(classes.btn, classes.cancelButton)} onClick={closePopup}>
+                        Cancel
+                    </button>
+                    <button
+                        id='extract-csv-button'
+                        type='button'
+                        disabled={!selectedFile}
+                        className={cx(
+                            classes.btn,
+                            classes.extractButton,
+                            !selectedFile && classes.disabledExtractButton
+                        )}
+                        onClick={handleExtractClick}
+                    >
+                        Extract Data
+                    </button>
+                </div>
+            </div>
+        </div>
+    )
+}
+
+export default UploadCSVPopup
+
+const useStyles = createUseStyles(theme => ({
+    fileIcon: {
+        width: '4rem',
+        height: '4rem',
+        margin: '20px 0',
```

Comment:
Instead of 20px can you use 1.25rem

Reply:
Absolutely. I'll get the margin fixed up and look into your file. It's definitely helpful having a more full csv like this, I hadn't seen any others in the project yet. I did put various errors in but will do my best to make sure that at least one error always shows up if the election obj doesn't get updated, and also will handle your case. Thanks!

Also sorry I wasn't able to join this past Tuesday! I was really hoping to but had something else that night.

Reply:
Looks like I had that working previously but then when I added some other stuff it must have broken. 
As part of the next component I may try to get jest working with storybook testing library so that there are a form of unit tests around all of this. It should help speed up development and ensure nothing breaks in the future too.
Just pushed a fix for all of the above! I did some more testing around it all and it seems to be good now. I added your csv file as a story titled "Other Data File" just so that it's recorded as well. Let me know if you see anything else I can update! 
(also just merged upstream main into this branch)